hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/PrecompileMessage.java

```diff
@@ -0,0 +1,229 @@
+		return htsOutputResult;
+	}
+
+	public long getGasRequired() {
+		return gasRequired;
+	}
+
+	public long getConsensusTime() {
+		return consensusTime.getEpochSecond();
+	}
+
+	public Bytes getInputData() {
+		return inputData;
+	}
+
+	public TokenID getTokenID() {
+		return tokenID;
+	}
+
+	public WorldLedgers getLedgers() {
+		return ledgers;
+	}
+
+	public long getGasRemaining() {
+		return gasRemaining;
+	}
+
+	public State getState() {
+		return state;
+	}
+
+	public Optional<Bytes> getRevertReason() {
+		return revertReason;
+	}
+
+	public List<Log> getLogs() {
+		return this.logs;
+	}
+
+	public Optional<ExceptionalHaltReason> getExceptionalHaltReason() {
+		return exceptionalHaltReason;
+	}
+
+	public void setRevertReason(Bytes revertReason) {
+		this.revertReason = Optional.ofNullable(revertReason);
+	}
+
+	public void setState(State state) {
+		this.state = state;
+	}
+
+	public void setGasRequired(long gasRequired) {
+		this.gasRequired = gasRequired;
+	}
+
+	public void setHtsOutputResult(Bytes htsOutputResult) {
+		this.htsOutputResult = htsOutputResult;
+	}
+
+	public void setExceptionalHaltReason(Optional<ExceptionalHaltReason> exceptionalHaltReason) {
+		this.exceptionalHaltReason = exceptionalHaltReason;
+	}
+
+	public void addLog(Log log) {
+		logs.add(log);
+	}
+
+	public void decrementRemainingGas(long amount) {
+		this.gasRemaining -= amount;
+	}
+
+	public byte[] unaliased(final byte[] evmAddress) {
+		final var addressOrAlias = Address.wrap(Bytes.wrap(evmAddress));
+		if (!addressOrAlias.equals(ledgers.canonicalAddress(addressOrAlias))) {
+			return new byte[20];
+		}
+		return ledgers.aliases().resolveForEvm(addressOrAlias).toArrayUnsafe();
+	}
+
+	public static class Builder {
+		private WorldLedgers ledgers;
+		private Address senderAddress;
+		private Wei value;
+		private Instant consensusTime;
+		private Long gasRemaining;
+		private Bytes inputData;
+		private TokenID tokenID;
+
+		private Builder() {
+		}
+
+		public Builder setLedgers(WorldLedgers ledgers) {
+			this.ledgers = ledgers;
+			return this;
+		}
+
+		public Builder setSenderAddress(Address senderAddress) {
+			this.senderAddress = senderAddress;
+			return this;
+		}
+
+		public Builder setValue(Wei value) {
+			this.value = value;
+			return this;
+		}
+
+		public Builder setConsensusTime(Instant consensusTime) {
+			this.consensusTime = consensusTime;
+			return this;
+		}
+
+		public Builder setGasRemaining(Long gasRemaining) {
+			this.gasRemaining = gasRemaining;
+			return this;
+		}
+
+		public Builder setInputData(Bytes inputData) {
+			this.inputData = inputData;
+			return this;
+		}
+
+		public Builder setTokenID(TokenID tokenID) {
+			this.tokenID = tokenID;
+			return this;
+		}
+
+
+		public PrecompileMessage build() {
+			return new PrecompileMessage(this.ledgers, this.senderAddress, this.value, this.consensusTime,
+					this.gasRemaining, this.inputData, this.tokenID);
+		}
+	}
+
+	public enum State {
```

Comment:
Imo we don't gain anything by redefining the `MessageFrame.State` enumeration here; would prefer to just re-use that type.